src/core.js

```diff
@@ -1457,37 +1472,54 @@ define([
         return true;
     };
         
-    fn.addQuestion = function (qType) {
+    fn.addQuestion = function (qType, position, refMug) {
         var _this = this,
             mug;
         this.ensureCurrentMugIsSaved(function () {
-            var foo = _this.getInsertTargetAndPosition(
-                _this.getCurrentlySelectedMug(), qType);
-            if (!foo) {
-                throw new Error("cannot add " + qType + " at the current position");
-            }
-            analytics.workflow("Added question in form builder");
-            mug = _this.data.core.form.createQuestion(foo.mug, foo.position, qType);
-
-            // Focus on first input, which might be a normal input or a rich text input
-            var $firstGroup = _this.$f.find(".fd-question-properties .form-group:first");
-            if ($firstGroup.length) {
-                var $input = $firstGroup.find("input, textarea");
-                if ($input.length) {
-                    // Rich text is off
-                    $input.focus();
-                } else {
-                    // Rich text is on
-                    $input = $firstGroup.find(".fd-textarea, .fd-input");
-                    richText.editor($input).focus();
+            if (position) {
+                if (!/^(before|after|into|first|last)$/.test(position)) {
+                    throw new Error("bad position: " + position);
+                }
+                if (!_this.isInsertAllowed(qType, position, refMug)) {
+                    throw new Error("cannot insert " + qType + " " + position +
+                        " " + (refMug ? refMug.hashtagPath : "root node"));
+                }
+            } else {
+                var foo = _this.getInsertTargetAndPosition(
+                    _this.getCurrentlySelectedMug(), qType);
+                if (!foo) {
+                    throw new Error("cannot add " + qType + " at the current position");
                 }
+                position = foo.position;
+                refMug = foo.mug;
             }
+            analytics.workflow("Added question in form builder");
```

Comment:
@krsdimagi I added analytics on command execution. It records the type of command (currently "add question" or "select question") that was executed. If the command could not be executed it records "bad command".